### PR TITLE
add back test itSendsTextShowPasswordToFrameworkOnAttach with new mock for display metrics

### PR DESCRIPTION
**Pull request by reidbaker**
Part 1/5 for re adding tests documented in flutter/flutter/issues/154746

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by matanlurey — APPROVED** [review]

**Comment by reidbaker** [comment]
Merging over the mac unopt test because it is likely https://github.com/flutter/flutter/issues/155110

**Comment by matanlurey** [comment]
> Merging over the mac unopt test because it is likely [flutter/flutter#155110](https://github.com/flutter/flutter/issues/155110)

While I don't think we have any particular rule against - I would caution against doing this in the future.

Let's escalate velocity issues with @jtmcdole and @johnmccutchan instead of "making our own velocity" by pushing on red.

**Comment by reidbaker** [comment]
> > Merging over the mac unopt test because it is likely [flutter/flutter#155110](https://github.com/flutter/flutter/issues/155110)
> 
> While I don't think we have any particular rule against - I would caution against doing this in the future.
> 
> Let's escalate velocity issues with @jtmcdole and @johnmccutchan instead of "making our own velocity" by pushing on red.

The tree has been too red this week to wait on green.

**Comment by matanlurey** [comment]
I agree - let's just make sure we're coupling actions like this with "here is an escalation that we can't do our jobs well and are being forced to make tradeoffs that are bad long-term".

File: shell/platform/android/test/README.md [hunk 1]
```diff
@@ -21,6 +21,10 @@ integration tests in other repos.
 4. Write your test.
 5. Build and run with `testing/run_tests.py [--type=java] [--java-filter=<test_class_name>]`.
 
+Example: from engine/src/flutter on a Mac
+`et build -c android_debug_unopt_arm64`
+`testing/run_tests.py --android-variant=android_debug_unopt_arm64 --type=java --java-filter=io.flutter.embedding.android.FlutterViewTest`
```
Comment (jtmcdole) [inline shell/platform/android/test/README.md:26]: would be great to have an `et test` command for this.

**Pull request merged**